pkg/virt-controller/watch/vmi_test.go

```diff
@@ -2476,6 +2476,89 @@ var _ = Describe("VirtualMachineInstance watcher", func() {
 				[]string{SuccessfulCreatePodReason}),
 		)
 
+		It("Should fail to get filesystem overhead if there are multiple CDI instances", func() {
+			cfg := cdiv1.CDIConfig{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "testConfig1",
+				},
+			}
+			cfg2 := cdiv1.CDIConfig{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "testConfig2",
+				},
+			}
+			// We add multiple CDI instances to trigger the error
+			cdiInformer.GetIndexer().Add(&cfg)
+			cdiInformer.GetIndexer().Add(&cfg2)
+
+			fsOverhead, err := controller.getFilesystemOverhead(nil)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal("more than one CDI resource detected: 2"))
+			Expect(fsOverhead).To(Equal(cdiv1.Percent("0")))
+		})
+
+		It("Should fail to get filesystem overhead if there is no CDI available", func() {
+			fsOverhead, err := controller.getFilesystemOverhead(nil)
+			Expect(err).To(HaveOccurred())
+			Expect(err.Error()).To(Equal(failedToFindCdi.Error()))
+			Expect(fsOverhead).To(Equal(cdiv1.Percent("0")))
+		})
+
+		It("Should fail to get filesystem overhead if there's no valid CDI config available", func() {
+			cfg := cdiv1.CDIConfig{
```

Comment:
you can see here that you supposedly add a config but its really should have been an instance, so failing of not having cdi config although you supposedly created here 1 is not quit right